### PR TITLE
Autumn Season

### PR DESCRIPTION
**Pull request by Flaborized**
## About the PR 

This PR adds some unique visuals and clothing during autumn, which is defined in the code as being all of September, October and November. Full list:

- Grass, trees and leaves are all changed to be more autumn-y. Mostly orange recolors, but leafygrass has a new sprite I drew for this.
- Four new clothing items added to the booth in an 'Autumn' category, only showing up during autumn itself (I plan on adding stuff for each season though)
- The clothing booth items include an autumn colored cape, a pumpkin themed jacket, a wearable tiny autumn tree, and a leaf wreath.
- Also the clown crate in the clown hole spawns with a special Autumn clown outfit, with a mask, hat, jumpsuit and shoes made to fit the season. I also intend on making one of these for each season as they come up
- All jumpsuits and suits have finished monkey versions

I might later make another PR that adds like, 'seasonal decoration' landmarks and sticks them in maps so we can have pumpkins and other seasonal stuff in appropriate spots (and then reuse those landmarks for other seasonal decor)

## Why's this needed? 

I think this is a cute idea! I've wanted to add more seasonal stuff for a while, this is meant to be more minor than spacemas/spooktober (it will overlap with spooktober even) and not really change anything mechanically, just a cute visual thing.

## Changelog 

```changelog
(u)Flaborized
(+)Autumn has come to the station, changing the plant life's hue and adding a few new items to the clothing booth for just the season. There's also a seasonal special clown costume in the clown hole crate!
```

### CODE REVIEW

**Comment by Glamurio** [comment]
Great idea, and not at all biased because autumn is my favorite season.

The only concern I have is the contrast between shrubs and leaves on ranches. Shrubs look great on regular grass, but the leaves make it look busy to the eyes. I love the leaf flooring, though it might need some tweaking, maybe.

File: code/obj/decorations.dm [hunk 1]
```diff
@@ -60,6 +60,12 @@
 	var/fallen = FALSE
 	var/fall_time = 2 SECONDS
 
+#ifdef AUTUMN
+	New()
+		icon_state = pick("tree_red", "tree_yellow", "tree_orange")
+		..()
```
Comment (pali6) [inline code/obj/decorations.dm:66]:
```suggestion
		..()
		icon_state = pick("tree_red", "tree_yellow", "tree_orange")
```
Call parent as the first statement in New please.

**Comment by Flaborized** [comment]
> Great idea, and not at all biased because autumn is my favorite season.
> 
> The only concern I have is the contrast between shrubs and leaves on ranches. Shrubs look great on regular grass, but the leaves make it look busy to the eyes. I love the leaf flooring, though it might need some tweaking, maybe.

Yeah I made the leaf texture a bit less saturated and darker to help with this somewhat, might require more tweaking though, will update the PR pictures and undraft in a bit after I write all the stuff out since I've added the stuff I wanna add to it

**Comment by Glamurio** [comment]
> Yeah I made the leaf texture a bit less saturated and darker to help with this somewhat, might require more tweaking though, will update the PR pictures and undraft in a bit after I write all the stuff out since I've added the stuff I wanna add to it

If the new pics are representative, I think it's a lot better now! The orange grass is also still in I presume? The one in botany, etc.

File: code/obj/item/clothing/masks.dm [hunk 2]
```diff
@@ -380,6 +380,12 @@ TYPEINFO(/obj/item/clothing/mask/monkey_translator)
 			src.item_state = "clown_hat"
 			user.show_text("You untuck the wig from the [src].")
 
+	autumn
+		name = "autumn clown wig and mask"
+		desc = "A special clown mask made to celebrate Autumn. Orange you glad you have it!!"
+		icon_state = "clown_autumn"
+		item_state = "clown_autumn"
```
Comment (Valtsu0) [inline code/obj/item/clothing/masks.dm:387]: Inconsistency bad. I do prefer `clown_autumn` over `autumn_clown` by a lot tho (kinda like color is always after).

File: code/obj/item/clothing/suits.dm [hunk 3]
```diff
@@ -2174,3 +2174,23 @@ TYPEINFO(/obj/item/clothing/suit/space/industrial/salvager)
...
+	body_parts_covered = TORSO
+	c_flags = ONBACK
+
+/obj/item/clothing/suit/autumn_jacket
+	name = "autumn jacket"
+	desc = "A jacket made to look like a pumpkin. It could just as easily be an orange though..."
+	icon = 'icons/obj/clothing/overcoats/item_suit_gimmick.dmi'
+	wear_image_icon = 'icons/mob/clothing/overcoats/worn_suit_gimmick.dmi'
+	icon_state = "autumn_jacket"
+	item_state = "autumn_jacket"
```
Comment (Valtsu0) [inline code/obj/item/clothing/suits.dm:2196]: This should be under `/obj/item/clothing/suit/jacket`

**Review by Valtsu0 — CHANGES_REQUESTED** [review]

File: code/modules/vending/clothingbooth_items.dm [hunk 4]
```diff
@@ -1227,4 +1227,37 @@ ABSTRACT_TYPE(/datum/clothingbooth_item/western/westboot)
 		name = "Brown Cowboy Boots"
 		path = /obj/item/clothing/shoes/westboot/brown
```
Comment (ZeWaka) [inline code/modules/vending/clothingbooth_items.dm:1229]: this file is so long :waa:

File: code/obj/item/clothing/masks.dm (hunk 2 above)
Comment (Flaborized) [reply code/obj/item/clothing/masks.dm:387]: did this (changed every other autumn clown item to use clown_autumn), though note that the way I had it set up was consistent *in the files they were in* so they're inconsistent with other clown items in that way. but then the clown items are inconsistent with themselves so :shrug:

File: code/obj/item/clothing/suits.dm [hunk 5]
```diff
@@ -2174,3 +2174,23 @@ TYPEINFO(/obj/item/clothing/suit/space/industrial/salvager)
 	icon_state = "fakebeewings"
 	item_state = "fakebeewings"
 	c_flags = ONBACK
+
+//Seasonal Stuff
+
+/obj/item/clothing/suit/autumn_cape
```
[the remark this answers is not in the document]
Comment (Flaborized) [reply code/obj/item/clothing/suits.dm:2185]: looks like bedsheet capes are jut clothing/suit so is similar yea

File: code/modules/vending/clothingbooth_items.dm (hunk 4 above)
Comment (Flaborized) [reply code/modules/vending/clothingbooth_items.dm:1315]: ☠️

**Review by Valtsu0 — APPROVED** [review]
Any remaining inconsistency is consistently inconsistent with the files they are in. I might consider fixing the inconsistent naming and insufficient datumization on a lot of things in clothing files after this gets merged, but it's not the job of this pr.

**Review by flappybatpal — CHANGES_REQUESTED** [review]

**Pull request merged**